Fix regression testing documentation

marchdf opened the pull request:
## Summary

This PR improves the documentation of regression testing, particularly comparisons with gold files (and generation of those). For the kind reviewer, it would be helpful to step through it as well to make sure I didn't forget anything.

## Pull request type

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. -->

Please check the type of change introduced:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## Additional background

Thanks @asalmgren for bringing this to my attention

mbkuhn commented:
Not sure what's going on with the failing Docs CI.

In my experience, when I build with spack, the AMR_WIND_TEST_WITH_FCOMPARE variable is on by default. When running the tests for golds, this is almost ok, but the problem is that none of the pass-dependent cases, i.e., the boundary plane input cases, will run, because all of the boundary plane output cases have failed. So then no golds are created for those cases. It might be best to comment on this variable some because I find it pretty important in my experience. 

Also, you might want to add a sentence more of detail explaining that when the tests complain that the plt file is missing, it means that the test simulation completed successfully but no comparison was made.

marchdf commented:
Figured out why the doc tests failed. 

Added a line re "explaining that when the tests complain that the plt file is missing"

Added something to the effect that you have to run it twice because of test dependencies.

I tried to stay away from any discussion/usage of spack since I think that would be confusing. That would require a whole different section I think.

mbkuhn reviewed (approved):
cheers to a documentation update!
